lib/seafoam/passes/truffle_translators/translators.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+require "set"
+
+module Seafoam
+  module Passes
+    module TruffleTranslators
+      autoload :Default, "seafoam/passes/truffle_translators/default"
+      autoload :TruffleRuby, "seafoam/passes/truffle_translators/truffleruby"
+
+      TRUFFLE_LANGUAGES = {
+        "org.truffleruby" => "TruffleRuby",
+      }
+
+      class << self
+        def get_translator(node)
+          queue = node.outputs.collect(&:to)
+          visited = Set.new
+
+          until queue.empty?
```

Comment:
Can we abstract this into a method which yields nodes, and one which looks at them?

Reply:
I've refactored a bit and extracted a BFS search search strategy to accompany a new `visit` and `visit_outputs` method pair in 8c535fe7724a58bc370647fefed47b599ed3cd8f. @eregon Do you think this satisfies the design change request?